Methods/Problem11/Program.cs

```diff
@@ -21,74 +21,74 @@ static void Main(string[] args)
+                {"2", "Option 2: Calculate the average of given sequence of numbers."},
+                {"3", "Option 3: Solve the linear equation a * x + b = 0"}
             };
 
-            foreach (string line in options)
+            foreach (var item in options)
             {
-                Console.WriteLine(line);
+                Console.WriteLine(item.Value);
             }
 
-            bool validation = false;
-            int option = 0;
-
-            while (!validation)
-            {
-                Console.WriteLine("Please enter option number:");
-                option = int.Parse(Console.ReadLine());
+            Console.WriteLine("Please enter option number:");
+            string option = Console.ReadLine();
 
-                validation = ValidateOption(option, options.Count);
-            }
+            ValidateOption(option, options);
 
             switch (option)
             {
-                case 1:
-                    string reversed = ReverseIntegarNumberOrder();
+                case "1":
+                    Console.WriteLine("Please enter integar to reverse:");
+                    string integer = Console.ReadLine();
+
+                    ValidateIntegarRange(integer);
+
+                    string reversed = ReverseIntegarNumberOrder(integer);
                     Console.WriteLine($"The reverse integar is {reversed}");
                     break;
 
-                case 2:
-                    double average = CalculateAverageOfSequence();
+                case "2":
+                    Console.WriteLine("Please enter sequence of numbers:");
+                    string sequence = Console.ReadLine();
+
+                    ValidateSequence(sequence);
+
+                    double average = CalculateAverageOfSequence(sequence);
                     Console.WriteLine($"The average value is {average}");
                     break;
 
-                case 3:
-                    double result = LinearEquation();
+                case "3":
+                    Console.WriteLine("Please enter coefficient a:");
+                    int a = int.Parse(Console.ReadLine());
+
+                    ValidateCoefficient(a);
+
+                    Console.WriteLine("Please enter coefficient b:");
+                    int b = int.Parse(Console.ReadLine());
+
+                    double result = LinearEquation(a, b);
                     Console.WriteLine($"The result is {result}");
                     break;
 
             }
         }
 
-        public static bool ValidateOption(int option, int acceptable)
+        public static bool ValidateOption
+            (string option, Dictionary<string, string> options)
         {
-            if (option < 1 || option > acceptable)
+            if (!options.ContainsKey(option))
             {
-                Console.WriteLine("Invalid option!");
-                return false;
+                throw new ArgumentException("Invalid Option!");
```

Comment:
Why do we need to throw Exception in validate function? Why do we need to pass the whole dictionary?